### PR TITLE
(Maybe) Fix Aliases

### LINKED ISSUE #516: R helpPanel not able to list packages for "Open help topic using ?"
**Describe the bug**
When clicking "Open help topic using `?`" an error is produced. 

Additionally, the error message advises that `r.helpPanel.rpath` ought to be set and pointing to a valid R executable. However, `r.helpPanel.rpath` is planned to be deprecated, so I'd imagine it should instead refer to `r.rpath.*` (in my case `* = mac`)

**To Reproduce**
1. Ensure all rpath settings are set to valid R executable
2. Open side panel
3. Click on "Open help topic using `?`"

setting.json

```json
  "r.bracketedPaste": true,
  "r.rpath.mac": "/usr/local/bin/R",
  "r.rterm.mac": "/Users/<me>/.local/bin/radian",
  "r.helpPanel.rpath": "/usr/local/bin/R",
  "r.sessionWatcher": true,
```

**Expected behavior**
Opens QuickPick with all available help aliases

**Screenshots**
You can show the keybord contents by pressing `F1` and `Developer: toggle screencast mode`
image

**Environment (please complete the following information):**
 - OS: macOS
 - VSCode Version: 1.52.1
 - R Version: 4.0.3
 - vscode-R version: 1.6.3

### PR DESCRIPTION
**Pull request by ManuelHentschel**
**What problem did you solve?**
This PR might fix #516 .

**How can I check this pull request?**
This PR changes the code used to retrieve a list of aliases from R. Now the aliases are written to a temporary file and read from there. Previously the aliases were transmitted using stdout, which could lead to a buffer overflow for large numbers of packages.

### CODE REVIEW

File: src/rHelpProvider.ts [hunk 1]
```diff
@@ -284,16 +291,26 @@ export class AliasProvider {
         this.allPackageAliases = null; 
 		const lim = '---vsc---'; // must match the lim used in R!
 		const re = new RegExp(`^.*?${lim}(.*)${lim}.*$`, 'ms');
-        const cmd = `${this.rPath} --silent --no-save --no-restore --slave -f "${this.rScriptFile}"`;
+        const tempDir = fs.mkdtempSync(path.join(os.tmpdir(), 'vscode-R-aliases-'));
+        const tempFile = path.join(tempDir, 'aliases.json');
+        const cmd = `${this.rPath} --silent --no-save --no-restore --slave -f "${this.rScriptFile}" > "${tempFile}"`;
```
Comment (renkun-ken) [inline src/rHelpProvider.ts:296]: Does this work on Windows?

File: src/rHelpProvider.ts [hunk 2]
```diff
@@ -284,16 +291,26 @@ export class AliasProvider {
         this.allPackageAliases = null; 
 		const lim = '---vsc---'; // must match the lim used in R!
 		const re = new RegExp(`^.*?${lim}(.*)${lim}.*$`, 'ms');
-        const cmd = `${this.rPath} --silent --no-save --no-restore --slave -f "${this.rScriptFile}"`;
+        const tempDir = fs.mkdtempSync(path.join(os.tmpdir(), 'vscode-R-aliases-'));
```
Comment (renkun-ken) [inline src/rHelpProvider.ts:294]: Is there a reason to create a temp dir rather than a simple temp file?

File: src/rHelpProvider.ts (hunk 1 above)
Comment (ManuelHentschel) [reply src/rHelpProvider.ts:296]: It does work on my windows machine, and the `>` operator seems to be part of cmd/powershell, so I think it should work.

File: src/rHelpProvider.ts (hunk 2 above)
Comment (ManuelHentschel) [reply src/rHelpProvider.ts:294]: Just the convenience of using `fs.mkdtempSync` to ensure we have a valid, unique path. Using a static filename might cause problems if multiple vscode instances are running simultaneously and generating a random filename (and checking its availability) would make things more complicated.

File: src/rHelpProvider.ts (hunk 1 above)
Comment (renkun-ken) [reply src/rHelpProvider.ts:296]: Thanks for your confirmation!

File: src/rHelpProvider.ts (hunk 2 above)
Comment (renkun-ken) [reply src/rHelpProvider.ts:294]: Thanks for the clarification!

**Review by renkun-ken — APPROVED** [review]
LGTM

**Pull request merged**